docs/book/src/clusterctl/developers.md

```diff
@@ -21,6 +19,34 @@ make clusterctl
 The output of the build is saved in the `bin/` folder; In order to use it you have to specify
 the full path, create an alias or copy it into a folder under your `$PATH`.
 
+## Use local artifacts
+
+Clusterctl by default uses artifacts published in the [providers repositories];
+during the development workflow it might happen you want to use artifacts from your local workstation.
+
+There are two options to do so:
+
+-  Use the [overrides layer], when you want to override a single published artifact with a local one.
+-  Create a local repository, when you want to avoid to use published artifacts and use intead the local ones.
+
+If you want to create a local artifact, follow those instructions:
+
+### Build artifacts locally
+
+In order to build artifacts for the CAPI core provider, the kubeadm bootstrap provider and the kubeadm control plane provider:
+
+```
+make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api
+make -C test/infrastructure/docker generate-manifests REGISTRY=gcr.io/k8s-staging-cluster-api
+```
+
+In order to build docker provider artifacts
+
+```
+make -C test/infrastructure/docker docker-build REGISTRY=gcr.io/k8s-staging-cluster-api
+make -C test/infrastructure/docker generate-manifests REGISTRY=gcr.io/k8s-staging-cluster-api
+```
+
```

Comment:
Since I can't mention this at the correct line, I'm doing it here. Since we are mentioning how to build docker provider artifacts, I thought it would make sense to include `infrastructure-docker` in the `clusterctl-settings.json` file.

```json
{
  "providers": ["cluster-api","bootstrap-kubeadm","control-plane-kubeadm", "infrastructure-aws", "infrastructure-docker"],	  
  "provider_repos": ["../cluster-api-provider-aws"]
}
```

Reply:
> For some reason, even after I loaded all the images with `kind load docker-image`, I keep getting `ImagePullBackOff. When I did a `kubectl describe pod -n capd-system ...` I got the following error:
> 
> ```
>   Warning  Failed       28m (x3 over 29m)      kubelet, kind-control-plane  Failed to pull image "gcr.io/k8s-staging-cluster-api/capd-manager:dev": rpc error: code = NotFound desc = failed to pull and unpack image "gcr.io/k8s-staging-cluster-api/capd-manager:dev": failed to resolve reference "gcr.io/k8s-staging-cluster-api/capd-manager:dev": gcr.io/k8s-staging-cluster-api/capd-manager:dev: not found
> ```

i had the same issue, the only fix that seemed to work for me was loading all the images (as you mention in the comments) and then changing the `imagePullPolicy` in all the deployments to `IfNotPresent`.

Reply:
done